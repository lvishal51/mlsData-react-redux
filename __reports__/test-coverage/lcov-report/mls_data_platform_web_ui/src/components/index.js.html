<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
