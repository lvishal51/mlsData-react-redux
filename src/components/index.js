export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
