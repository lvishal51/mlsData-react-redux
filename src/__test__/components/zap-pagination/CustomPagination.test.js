import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { CustomPagination } from 'components';

const pageData = [
  {
    "id": 1,
    "name": "ADDRESS",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  },
  {
    "id": 2,
    "name": "AGE",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  },
  {
    "id": 3,
    "name": "AMENITIES",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  },
  {
    "id": 4,
    "name": "AREA",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": "This field have to be a INTEGER ",
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  },
  {
    "id": 5,
    "name": "ASSESSMENTS",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  }
];

const updatedPageData = [
  {
    "id": 1,
    "name": "ADDRESS",
    "type": "REPORT",
    "dataType": "MEDIUMTEXT",
    "dataLength": 0,
    "notes": null,
    "lastUpdateTs": "2018-10-15T18:30:00.000Z",
    "createTs": null
  }
]

const pageContent = {
    payload: pageData
  }

const inputProps = {
  headerText: "Canonical Fields",
  numberOfEntriesPerPage: 1,
  pageContent: pageContent,
  keyName: "name",
  onItemClick: () => {},
  showAddNewSection: true,
  addNewValue: () => {},
  resetPagination: false
};

describe('Zap Pagination component', () => {
  it('should render correctly.', () => {
    const output = shallow(<CustomPagination {...inputProps} />);
    output.setProps({pageData: updatedPageData});
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should render correctly.', () => {
    const output = shallow(<CustomPagination {...inputProps} numberOfEntriesPerPage={5} />);
    output.setState({pageNumber: 1, selectedValueId: 1});
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('onNextClick function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onNextClick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(6).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onNextClick function when user is on last page', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    wrapper.setState({pageNumber: 5});
    const spy = jest.spyOn(wrapper.instance(), "onNextClick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(6).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onPageNumberclick function at first page', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onPageNumberclick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(1).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onPageNumberclick function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onPageNumberclick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(3).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onPageNumberclick function at last page', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onPageNumberclick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(5).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onPreviousClick function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onPreviousClick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(0).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onPreviousClick function hen user clicked on in between page', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    wrapper.setState({pageNumber: 3})
    const spy = jest.spyOn(wrapper.instance(), "onPreviousClick");
    wrapper.instance().forceUpdate();
    wrapper.find("PaginationLink").at(0).simulate('click');
    expect(spy).toBeCalled();
  });

  it('clearFilter function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "clearFilter");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(2).simulate('click');
    expect(spy).toBeCalled();
  });

  it('clearFilter function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    wrapper.setState({searchQuery: 's'});
    const spy = jest.spyOn(wrapper.instance(), "clearFilter");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(2).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onCombinedSearch function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const spy = jest.spyOn(wrapper.instance(), "onCombinedSearch");
    wrapper.instance().forceUpdate();
    wrapper.find("ListGroupItem").at(1).simulate('click');
    expect(spy).toBeCalled();
  });

  it('onSearchText function should be triggered correctly', () => {
    const wrapper = shallow(<CustomPagination {...inputProps} />);
    const event = { preventDefault() {}, target: { value: 'A' } };
    const spy = jest.spyOn(wrapper.instance(), "onSearchText");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('change', event);
    expect(spy).toBeCalled();
  });

});
