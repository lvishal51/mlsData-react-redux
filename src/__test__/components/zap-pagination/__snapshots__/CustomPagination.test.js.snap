// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Zap Pagination component should render correctly. 1`] = `
<Card
  tag="div"
>
  <CardHeader
    className="customCardHeader"
    tag="div"
  >
    <span
      className="h5"
    >
      Canonical Fields
    </span>
    <Button
      className="zapButton"
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-plus-circle"
      />
      <span
        className="sr-only"
      >
        Add new value
      </span>
    </Button>
  </CardHeader>
  <CardBody
    className="customCardBody"
    tag="div"
  >
    <FormGroup
      tag="div"
    >
      <DebounceInput
        className="form-control"
        debounceTimeout={300}
        element="input"
        forceNotifyByEnter={true}
        forceNotifyOnBlur={true}
        minLength={0}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="search"
      />
    </FormGroup>
    <div
      className="d-flex mt-1"
    >
      <PaginationItem
        disabled={true}
        tag="span"
      >
        <PaginationLink
          onClick={[Function]}
          previous={true}
          tag="a"
        />
      </PaginationItem>
      <Pagination
        aria-label="pagination"
        className="customPagination"
        listTag="ul"
        tag="nav"
      >
        <PaginationItem
          key="0"
          tag="li"
        >
          <PaginationLink
            className="activePage"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            1
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="1"
          tag="li"
        >
          <PaginationLink
            className=""
            key="1"
            onClick={[Function]}
            tag="a"
          >
            2
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="2"
          tag="li"
        >
          <PaginationLink
            className=""
            key="2"
            onClick={[Function]}
            tag="a"
          >
            3
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="3"
          tag="li"
        >
          <PaginationLink
            className=""
            key="3"
            onClick={[Function]}
            tag="a"
          >
            4
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="4"
          tag="li"
        >
          <PaginationLink
            className=""
            key="4"
            onClick={[Function]}
            tag="a"
          >
            5
          </PaginationLink>
        </PaginationItem>
      </Pagination>
      <PaginationItem
        disabled={false}
        tag="span"
      >
        <PaginationLink
          href="javascript:void(0);"
          next={true}
          onClick={[Function]}
          tag="a"
        />
      </PaginationItem>
    </div>
    <Row
      className="paginationCard"
      tag="div"
    >
      <Col
        md="10"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <ListGroup
          flush={true}
          tag="ul"
        >
          <ListGroupItem
            className="d-flex justify-content-between text-dark"
            href="javascript:void(0);"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            <span>
              ADDRESS
            </span>
          </ListGroupItem>
        </ListGroup>
      </Col>
      <Col
        md="2"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <ListGroup
          className="letterList"
          flush={true}
          tag="ul"
        >
          <ListGroupItem
            className=""
            href="javascript:void(0);"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            A
          </ListGroupItem>
          <ListGroupItem
            className="clearFilter fa fa-times-circle"
            href="javascript:void(0);"
            onClick={[Function]}
            tag="a"
            title="Clear"
          />
        </ListGroup>
      </Col>
    </Row>
    <div
      className="d-flex mt-1"
    >
      <PaginationItem
        disabled={true}
        tag="span"
      >
        <PaginationLink
          onClick={[Function]}
          previous={true}
          tag="a"
        />
      </PaginationItem>
      <Pagination
        aria-label="pagination"
        className="customPagination"
        listTag="ul"
        tag="nav"
      >
        <PaginationItem
          key="0"
          tag="li"
        >
          <PaginationLink
            className="activePage"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            1
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="1"
          tag="li"
        >
          <PaginationLink
            className=""
            key="1"
            onClick={[Function]}
            tag="a"
          >
            2
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="2"
          tag="li"
        >
          <PaginationLink
            className=""
            key="2"
            onClick={[Function]}
            tag="a"
          >
            3
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="3"
          tag="li"
        >
          <PaginationLink
            className=""
            key="3"
            onClick={[Function]}
            tag="a"
          >
            4
          </PaginationLink>
        </PaginationItem>
        <PaginationItem
          key="4"
          tag="li"
        >
          <PaginationLink
            className=""
            key="4"
            onClick={[Function]}
            tag="a"
          >
            5
          </PaginationLink>
        </PaginationItem>
      </Pagination>
      <PaginationItem
        disabled={false}
        tag="span"
      >
        <PaginationLink
          href="javascript:void(0);"
          next={true}
          onClick={[Function]}
          tag="a"
        />
      </PaginationItem>
    </div>
  </CardBody>
</Card>
`;

exports[`Zap Pagination component should render correctly. 2`] = `
<Card
  tag="div"
>
  <CardHeader
    className="customCardHeader"
    tag="div"
  >
    <span
      className="h5"
    >
      Canonical Fields
    </span>
    <Button
      className="zapButton"
      color="primary"
      onClick={[Function]}
      tag="button"
    >
      <i
        aria-hidden="true"
        className="fa fa-plus-circle"
      />
      <span
        className="sr-only"
      >
        Add new value
      </span>
    </Button>
  </CardHeader>
  <CardBody
    className="customCardBody"
    tag="div"
  >
    <FormGroup
      tag="div"
    >
      <DebounceInput
        className="form-control"
        debounceTimeout={300}
        element="input"
        forceNotifyByEnter={true}
        forceNotifyOnBlur={true}
        minLength={0}
        name="search"
        onChange={[Function]}
        placeholder="Search"
        type="search"
      />
    </FormGroup>
    <Row
      className="paginationCard"
      tag="div"
    >
      <Col
        md="10"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <ListGroup
          flush={true}
          tag="ul"
        >
          <ListGroupItem
            className="d-flex justify-content-between text-dark selectedValue"
            href="javascript:void(0);"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            <span>
              ADDRESS
            </span>
          </ListGroupItem>
          <ListGroupItem
            className="d-flex justify-content-between text-dark"
            href="javascript:void(0);"
            key="1"
            onClick={[Function]}
            tag="a"
          >
            <span>
              AGE
            </span>
          </ListGroupItem>
          <ListGroupItem
            className="d-flex justify-content-between text-dark"
            href="javascript:void(0);"
            key="2"
            onClick={[Function]}
            tag="a"
          >
            <span>
              AMENITIES
            </span>
          </ListGroupItem>
          <ListGroupItem
            className="d-flex justify-content-between text-dark"
            href="javascript:void(0);"
            key="3"
            onClick={[Function]}
            tag="a"
          >
            <span>
              AREA
            </span>
          </ListGroupItem>
          <ListGroupItem
            className="d-flex justify-content-between text-dark"
            href="javascript:void(0);"
            key="4"
            onClick={[Function]}
            tag="a"
          >
            <span>
              ASSESSMENTS
            </span>
          </ListGroupItem>
        </ListGroup>
      </Col>
      <Col
        md="2"
        tag="div"
        widths={
          Array [
            "xs",
            "sm",
            "md",
            "lg",
            "xl",
          ]
        }
        xs="12"
      >
        <ListGroup
          className="letterList"
          flush={true}
          tag="ul"
        >
          <ListGroupItem
            className=""
            href="javascript:void(0);"
            key="0"
            onClick={[Function]}
            tag="a"
          >
            A
          </ListGroupItem>
          <ListGroupItem
            className="clearFilter fa fa-times-circle"
            href="javascript:void(0);"
            onClick={[Function]}
            tag="a"
            title="Clear"
          />
        </ListGroup>
      </Col>
    </Row>
  </CardBody>
</Card>
`;
